Firestore/core/src/firebase/firestore/model/precondition.h

```diff
@@ -131,9 +131,7 @@ class Precondition {
             stringWithFormat:@"<Precondition update_time=%s>",
                              update_time_.timestamp().ToString().c_str()];
     }
-    // We only raise dev assertion here. This function is mainly used in
-    // logging.
-    FIREBASE_DEV_ASSERT_MESSAGE(false, "precondition invalid");
+    HARD_FAIL("precondition invalid");
```

Comment:
A few notes/questions:
1) Line 135 is now unreachable. You could optionally remove it. Alternatively, you could remove the assert (and replace with a log) since the old comment indicated the dev assert was mostly for logging purposes.
2) Assuming type_ is an enum, does the above switch exhaustively check all values of the enum? If so, this could change to FIREBASE_UNREACHABLE(). (Given our compiler flags, the compiler should warn/error if we add more values to the enum.)

Reply:
It's an enum. Changed to UNREACHABLE.